src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php

```diff
@@ -181,12 +178,9 @@ public function configureOptions(OptionsResolver $resolver)
         // are indexed by an incrementing integer.
         // Use the ID/incrementing integer as choice value.
         $choiceValue = function (Options $options) {
-            /** @var IdReader $idReader */
-            $idReader = $options['id_reader'];
-
             // If the entity has a single-column ID, use that ID as value
-            if ($idReader->isSingleId()) {
-                return [$idReader, 'getIdValue'];
+            if ($options['id_reader'] instanceof IdReader && $options['id_reader']->isSingleId()) {
```

Comment:
(optional) If we trust the normalizer, maybe we can get rid of `instanceof` everywhere and compare with` null`.

Reply:
It helps autocomplete while getting rid of the duck type hinting I've removed in the process.